### DISCUSSION TITLE
TypeScript issues when updating data

### DISCUSSION THREAD

**Discussion by harblaith7** [nuxt/nuxt#17841] [Questions] [answered]
I am having issues updating state with typescript.

If I set my state in my setup method like so
```typescript
let name = useState<string>("name", () => "");
```

Then create method to update it
```javascript
 const updateName = () => {
    name = "Laith"
 }
```

I get the following typescript error 

`Type 'string' is not assignable to type 'Ref<string>'.`

Any idea how to fix this?

**Comment by psycongaroo** [comment, marked as answer]
```
 const updateName = () => {
    name.value = "Laith"
 }
```
